Realm/RLMResults.mm

```diff
@@ -286,11 +286,15 @@ - (NSUInteger)indexOfObjectWithPredicate:(NSPredicate *)predicate {
 
     Query query = translateErrors([&] { return _results.get_query(); });
     RLMUpdateQueryWithPredicate(&query, predicate, _realm.schema, _objectSchema);
-    size_t index = query.find();
-    if (index == realm::not_found) {
+
+    // FIXME: We're only looking for a single object so we'd like to be able to use `Query::find`
+    // for this, but as of core v0.97.1 it gives incorrect results if the query is restricted
+    // to a link view (<https://github.com/realm/realm-core/issues/1565>).
+    auto table_view = query.find_all();
```

Comment:
Oh, good point! That'd just give us the overhead of working with the `TableView`, which at least will be fixed rather than dependent on the number of rows we're searching.